XErrorDB fixes applied to X.Org git after libX11-1.1.3 was released.
These should all be included in libX11-1.1.4 so this patch can be
removed when upgrading to that release.

From 31540f1438ec63faf37044f2fd654b335ddf80f0 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Kristian_H=C3=B8gsberg?= <krh@redhat.com>
Date: Wed, 29 Aug 2007 19:50:57 -0400
Subject: [PATCH] Add GLX 1.4 requests and errors.
---
 src/XErrorDB |   16 ++++++++++++++++
 1 files changed, 16 insertions(+), 0 deletions(-)

diff --git a/src/XErrorDB b/src/XErrorDB
index 2e34b1d..a0b9684 100644
--- a/src/XErrorDB
+++ b/src/XErrorDB
@@ -498,6 +498,18 @@ XRequest.GLX.17: X_GLXVendorPrivateWithR
 XRequest.GLX.18: X_GLXQueryExtensionsString
 XRequest.GLX.19: X_GLXQueryServerString
 XRequest.GLX.20: X_GLXClientInfo
+XRequest.GLX.21: X_GLXGetFBConfigs
+XRequest.GLX.22: X_GLXCreatePixmap
+XRequest.GLX.23: X_GLXDestroyPixmap
+XRequest.GLX.24: X_GLXCreateNewContext
+XRequest.GLX.25: X_GLXQueryContext
+XRequest.GLX.26: X_GLXMakeContextCurrent
+XRequest.GLX.27: X_GLXCreatePbuffer
+XRequest.GLX.28: X_GLXDestroyPbuffer
+XRequest.GLX.29: X_GLXGetDrawableAttributes
+XRequest.GLX.30: X_GLXChangeDrawableAttributes
+XRequest.GLX.31: X_GLXCreateWindow
+XRequest.GLX.32: X_GLXDestroyWindow
 ! GL Non-rendering Commands
 XRequest.GLX.103: X_GLXDeleteLists
 XRequest.GLX.102: X_GLXEndList
@@ -555,6 +567,10 @@ XProtoError.GLX.5: GLXBadCurrentWindow
 XProtoError.GLX.6: GLXBadRenderRequest
 XProtoError.GLX.7: GLXBadLargeRequest
 XProtoError.GLX.8: GLXUnsupportedPrivateRequest
+XProtoError.GLX.9: GLXBadFBConfig
+XProtoError.GLX.10: GLXBadPbuffer
+XProtoError.GLX.11: GLXBadCurrentDrawable
+XProtoError.GLX.12: GLXBadWindow
 ! XC-MISC extension, an X Consortium standard
 XRequest.XC-MISC.0: XCMiscGetVersion
 XRequest.XC-MISC.1: XCMiscGetXIDRange
-- 
1.4.1

From 416a812200f24d19149dcc497e5c51a0608120f6 Mon Sep 17 00:00:00 2001
From: Alan Coopersmith <alan.coopersmith@sun.com>
Date: Fri, 8 Feb 2008 15:31:31 -0800
Subject: [PATCH] XErrorDB updates for Render 0.9 & XFixes 4.0
---
 src/XErrorDB |   10 +++++++++-
 1 files changed, 9 insertions(+), 1 deletions(-)

diff --git a/src/XErrorDB b/src/XErrorDB
index a0b9684..70715a5 100644
--- a/src/XErrorDB
+++ b/src/XErrorDB
@@ -634,7 +634,7 @@ XProtoError.XFree86-Misc.5: XF86MiscModI
 XProtoError.XFree86-Misc.6: XF86MiscModInDevClientNotLocal
 XProtoError.XFree86-Misc.7: XF86MiscNoModule
 ! XFree86-DGA  -- not an X Consortium standard
-XRequest.XFree86-DGA.0: XDGAQueryVersion
+XRequest.XFree86-DGA.0: XF86DGAQueryVersion
 XRequest.XFree86-DGA.1: XF86DGAGetVideoLL
 XRequest.XFree86-DGA.2: XF86DGADirectVideo
 XRequest.XFree86-DGA.3: XF86DGAGetViewPortSize
@@ -891,6 +891,12 @@ XRequest.RENDER.27: RenderCreateCursor
 XRequest.RENDER.28: RenderSetPictureTransform
 XRequest.RENDER.29: RenderQueryFilters
 XRequest.RENDER.30: RenderSetPictureFilter
+XRequest.RENDER.31: RenderCreateAnimCursor
+XRequest.RENDER.32: RenderAddTraps
+XRequest.RENDER.33: RenderCreateSolidFill
+XRequest.RENDER.34: RenderCreateLinearGradient
+XRequest.RENDER.35: RenderCreateRadialGradient
+XRequest.RENDER.36: RenderCreateConicalGradient
 XProtoError.RENDER.0: RenderBadFormat (invalid PictFormat parameter)
 XlibMessage.RENDER.0: PictFormat id in failed request: 0x%lx
 XProtoError.RENDER.1: RenderBadPicture (invalid Picture parameter)
@@ -931,6 +937,8 @@ XRequest.XFIXES.25: XFixesGetCursorImage
 XRequest.XFIXES.26: XFixesChangeCursor
 XRequest.XFIXES.27: XFixesChangeCursorByName
 XRequest.XFIXES.28: XFixesExpandRegion
+XRequest.XFIXES.29: XFixesHideCursor
+XRequest.XFIXES.30: XFixesShowCursor
 XProtoError.XFIXES.0: BadRegion (invalid Region parameter)
 XlibMessage.XFIXES.0: Region id in failed request:  0x%lx
 ! DAMAGE extension
-- 
1.4.1

